includes/class-magic-link.php

```diff
@@ -346,11 +348,15 @@ public static function generate_token( $user ) {
 		if ( ! empty( $tokens ) ) {
 			/** Limit maximum tokens to 5. */
 			$tokens = array_slice( $tokens, -4, 4 );
-			/** Clear expired tokens. */
 			foreach ( $tokens as $index => $token_data ) {
+				/** Clear expired tokens. */
 				if ( $token_data['time'] < $expire ) {
 					unset( $tokens[ $index ] );
 				}
+				/** Rate limit token generation. */
+				if ( $token_data['time'] + self::RATE_INTERVAL > $now ) {
+					return new \WP_Error( 'rate_limit_exceeded', __( 'You must wait before you can issue another authorization code.', 'newspack' ) );
```

Comment:
Language tweak suggestion:

```suggestion
					return new \WP_Error( 'rate_limit_exceeded', __( 'Please wait a minute before requesting another authorization code.', 'newspack' ) );
```